### PR TITLE
Fix error in User_Memory_Init_Pools

### PR DESCRIPTION
**Pull request by xezon**
* Split off of #796

This claims to be a fix for a Thyme specific issue.

### CODE REVIEW

File: src/w3d/renderer/ddsfile.cpp [hunk 1]
```diff
@@ -592,6 +592,7 @@ bool DDSFileClass::Load()
 
     if (data == 0) {
         captainslog_warn("DDSFile '%s' appears to contain no data, size is %d.", m_name, fp->Size());
+        fp->Close();
```
Comment (tomsons26) [inline src/w3d/renderer/ddsfile.cpp]: auto_file_ptr should indirectly be doing that?

Comment (jonwil) [reply src/w3d/renderer/ddsfile.cpp]: No it does not.

Comment (tomsons26) [reply src/w3d/renderer/ddsfile.cpp]: right, comment as bugfix

Comment (xezon) [reply src/w3d/renderer/ddsfile.cpp]: Perhaps the right fix here is then to auto close on destruction of `auto_file_ptr` instead.

I will take a look if there are more places where `auto_file_ptr` is missing call to Close.

Comment (jonwil) [reply src/w3d/renderer/ddsfile.cpp]: Original auto_file_ptr didn't call Close, we should keep it that way.

Comment (xezon) [reply src/w3d/renderer/ddsfile.cpp]: I have reverted change to DDSFileClass::Load and will make a new Pull for `auto_file_ptr`

Comment (xezon) [reply src/w3d/renderer/ddsfile.cpp]: I have taken another look at `auto_file_ptr` and `GameFileClass` in debugger. It does call `Close` on destruction via virtual destructor:

```c++
GameFileClass::~GameFileClass()
{
    Close();
}
```

So this needs no change.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**